Birthday-2024-Project/Scripts/GridLogic.gd

```diff
@@ -1,17 +1,39 @@
 extends Node2D
+class_name GridLogic
+
+const MAX_HEIGHT = 20
+const MAX_WIDTH = 20
 
 #have this passed by the level selector in the future
 @export var debug_setupData : LevelSetup
 
 var availablePieces : Array[Node2D]
 
 #var 2D array of grid occupancies
+var gridSpaces = []
```

Comment:
I think it would be good to implement static typing right from the beginning since it will help with performance and error detection / prevention.
so here : `var gridSpaces: Array = []`